frontend/src/metabase/query_builder/components/CompareAggregations/utils.ts

```diff
@@ -91,6 +137,71 @@ export const canSubmit = (
   return isPeriodValid && areColumnsValid;
 };
 
+type UpdatedQuery = {
+  query: Lib.Query;
+  addedAggregations: Lib.ExpressionClause[];
+};
+
+export function updateQueryWithCompareOffsetAggregations(
+  query: Lib.Query,
+  stageIndex: number,
+  aggregation: Lib.AggregationClause | Lib.ExpressionClause,
+  offset: "" | number,
+  columns: ColumnType[],
+  columnAndBucket: BreakoutColumnAndBucket,
+): UpdatedQuery | null {
+  if (!aggregation || offset === "" || !columnAndBucket) {
```

Comment:
Types already account for this

```suggestion
  if (!aggregation || offset === "") {
```